engineapi/instance_manager.go

```diff
@@ -240,10 +240,22 @@ func parseProcess(p *imapi.Process) *longhorn.InstanceProcess {
 			Conditions: p.ProcessStatus.Conditions,
 			PortStart:  p.ProcessStatus.PortStart,
 			PortEnd:    p.ProcessStatus.PortEnd,
+
+			// FIXME: These fields are not used, maybe we can deprecate them later.
+			Listen:   "",
+			Endpoint: "",
 		},
 	}
 }
 
+func getDataEngineFromInstanceProcess(p *imapi.Instance) longhorn.DataEngineType {
+	if p.DataEngine != "" {
+		return longhorn.DataEngineType(p.DataEngine)
+	}
+
+	return longhorn.DataEngineType(p.DataEngine)
```

Comment:
This returns the same value as the condition above?

Reply:
Good :eye:

Reply:
The original one is correct. That is falling back to backendStoreDriver.
static checker does not like reading data from deprecated field...

Reply:
Any solution to the golinter's deprecation check?
@innobead @c3y1huang

Reply:
I think we could ignore `SA1019`? 
https://github.com/longhorn/longhorn-manager/pull/2400/files#diff-4c0e1e708c59d9a06da4c01a46c2d30739fce6b57e17eff33b2930da8d294692R25

Reply:
Can we ignore a specific line? I feel it should be feasible.

Reply:
https://golangci-lint.run/usage/false-positives/#nolint-directive

Reply:
Not sure why `// lint:ignore SA1019` doesn't work. Let me try `nolint`

Reply:
`nolint: all` works